standards/chronologic-piper.md

```diff
@@ -0,0 +1,124 @@
+# Specification
+
+## Summary
+A standardized interface for interacting with subscriptions.
+
+##  Abstract
+This standard defines a contract interface for interacting with subscription data. The scope of this proposal is end-user / wallet interactions. Creation of the subscriptions is out-of-scope of this ERC.
+
+## Our Thoughts
+From the subscriber's side:
+
+* I want strong guarantees on when I can cancel my subscription.
+* I will normally want payments to happen automatically without any action on my part.
+* In some cases it might be valuable to require an approval process.
+* For dynamically priced subscriptions I want to be able to set limits (require authorization if subscription is more than X).
+
+
+From the providers's side:
+
+* I need the ability to charge a fixed fee per subscription time unit (netflix, pandora, etc).
+* I need the ability to charge a dynamic fee per subscription time unit (aws, twilio, etc).
+* I need to be able to create reasonably accurate forecasts for upcoming subscriptions: Programatic checks that subscription accounts have available balance and that subscription is active.
+
+## Motivation
+Standard that covers all usecases of subscription based model with minimal interface suitable for wallets.
+
+### Interface
+
+#### cancel
+
+Immediately cancels the subscription. Implementations should ensure that any unpaid subscription payments are paid to the provider as part of cancellation to ensure providers are able to let subscriptions fees fill up for arbitrary lengths of time, allowing them to reduce overhead from transaction costs.
+
+```SOLIDITY
+
+/** @dev Cancels the subscription
+  * @return true if cancellation succedded 
+*/
+
+function cancel()
+    public
+    returns (bool success);
+```
+
+#### isFunded
+
+Checkes if there is enough funds available for the provider to collect
+ 
+```SOLIDITY
+
+/** @dev Cancels the subscription
+  * @return true if there is enough funds allowed to the contract to pay for subscription 
+*/
+
+function isFunded()
+    public
+    returns (bool funded);
+```
+
+#### availableFunds
+
+Returns the amount of ETH or tokens available for the provider to collect
+
+ 
+```SOLIDITY
+
+/** @dev The amount of ETH or tokens available for the provider
+  * @return available funds
+*/
+
+function availableFunds()
+    public
+    returns (uint256 funds);
+```
+
+#### metadata
```

Comment:
I think this is a good idea. Can we agree on this more polished version: https://github.com/EthereumOpenSubscriptions/standard/blob/99d833792ddd4d2a422fea66fb21379c7a2ef61a/standards/stef_and_luka-standard_proposal_draft.md#erc948metadata-interface? It was taken directly from ERC721 standard and I think they did a good job with the spec.

Reply:
This is a very good point!

What are your thoughts on storing metadata as JSON key-value object?

Reply:
I like this idea. Removes some pollution caused by implementation specifics.